Solution template for WEEKLY ML CODE CLUB: 001

This file is Adesoba Olamide solution for the WEEKLY ML CODE CLUB: 001. It will walk you through my analysis of the tasks

In [1]:
# loading all libraries required for this tasks
import pandas as pd
import numpy as np  

Exploratory data analysis The first step in any analysis is to first understand the data. Let's take a look at each of the datasets provided.

Loading the  required  datasets and assigning a variable name
WHERE CP = CELL_PHONES DATA
        P = POPULATION DATA 
        GC = COUNTRIES GEOGRAPHY

In [2]:
CP = pd.read_csv(r'C:\Users\OLAMIDE\Desktop\HAMOYE\WEEK1 CODE\cell_phones.csv')
P = pd.read_csv(r'C:\Users\OLAMIDE\Desktop\HAMOYE\WEEK1 CODE\population1.csv')
GC = pd.read_csv(r'C:\Users\OLAMIDE\Desktop\HAMOYE\WEEK1 CODE\ddf--entities--geo--country.csv')

In [4]:
CP.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 11640 entries, 0 to 11639
Data columns (total 3 columns):
 #   Column             Non-Null Count  Dtype  
---  ------             --------------  -----  
 0   Country            11640 non-null  object 
 1   year               11640 non-null  int64  
 2   cell_phones_total  11640 non-null  float64
dtypes: float64(1), int64(1), object(1)
memory usage: 272.9+ KB


In [7]:
CP.isnull().sum()

Country              0
year                 0
cell_phones_total    0
dtype: int64

In [10]:
CP.head()

Country  year  cell_phones_total
0  Afghanistan  1960                0.0
1  Afghanistan  1961                0.0
2  Afghanistan  1962                0.0
3  Afghanistan  1963                0.0
4  Afghanistan  1964                0.0

In [11]:
CP.describe()

year  cell_phones_total
count  11640.000000       1.164000e+04
mean    1989.500000       7.875722e+06
std       17.318846       5.572496e+07
min     1960.000000       0.000000e+00
25%     1974.750000       0.000000e+00
50%     1989.500000       0.000000e+00
75%     2004.250000       5.615000e+05
max     2019.000000       1.730000e+09

The cell phone data is clean i.e it is devoid of null value, with 11640 rows and 4 columns

In [12]:
P.info()

<class 'pandas.core.frame.DataFrame'>
RangeIndex: 58695 entries, 0 to 58694
Data columns (total 3 columns):
 #   Column      Non-Null Count  Dtype 
---  ------      --------------  ----- 
 0   Country     58695 non-null  object
 1   year        58695 non-null  int64 
 2   population  58695 non-null  int64 
dtypes: int64(2), object(1)
memory usage: 1.3+ MB


In [13]:
P.isnull().sum()

Country       0
year          0
population    0
dtype: int64

In [14]:
P.head()

Country  year  population
0  Afghanistan  1800     3280000
1  Afghanistan  1801     3280000
2  Afghanistan  1802     3280000
3  Afghanistan  1803     3280000
4  Afghanistan  1804     3280000

In [15]:
P.describe()

year    population
count  58695.000000  5.869500e+04
mean    1950.000000  2.328525e+07
std       86.891476  1.007173e+08
min     1800.000000  6.450000e+02
25%     1875.000000  4.220000e+05
50%     1950.000000  2.610000e+06
75%     2025.000000  1.080000e+07
max     2100.000000  1.650000e+09

The Population data is clean i.e it is devoid of null value, with 58695 rows and 3 columns

In [24]:
CP_P = CP.merge(P, how='inner', on=['Country','year'])
np.shape(CP_P)

(11640, 4)

In [90]:
GC = GC.rename(columns={'country': 'Geo','name':'Country'})
NEWGC = GC[['Geo','Country']]
NEWGC

Geo               Country
0         abkh              Abkhazia
1          abw                 Aruba
2          afg           Afghanistan
3          ago                Angola
4          aia              Anguilla
..         ...                   ...
268  yem_south  South Yemen (former)
269        yug            Yugoslavia
270        zaf          South Africa
271        zmb                Zambia
272        zwe              Zimbabwe

[273 rows x 2 columns]

In [91]:
CP_P_GC = CP_P.merge(NEWGC, how='inner', on='Country')
CP_P_GC

Country  year  cell_phones_total  population  Geo
0      Afghanistan  1960                0.0     9000000  afg
1      Afghanistan  1961                0.0     9170000  afg
2      Afghanistan  1962                0.0     9350000  afg
3      Afghanistan  1963                0.0     9540000  afg
4      Afghanistan  1964                0.0     9740000  afg
...            ...   ...                ...         ...  ...
11635     Zimbabwe  2015         12800000.0    13800000  zwe
11636     Zimbabwe  2016         12900000.0    14000000  zwe
11637     Zimbabwe  2017         14100000.0    14200000  zwe
11638     Zimbabwe  2018         12900000.0    14400000  zwe
11639     Zimbabwe  2019         13200000.0    14600000  zwe

[11640 rows x 5 columns]

In [92]:
CP_P_GC['CELL_PHONE_PER_PERSON'] = CP_P_GC['cell_phones_total']/CP_P_GC['population']
CP_P_GC

Country  year  cell_phones_total  population  Geo  \
0      Afghanistan  1960                0.0     9000000  afg   
1      Afghanistan  1961                0.0     9170000  afg   
2      Afghanistan  1962                0.0     9350000  afg   
3      Afghanistan  1963                0.0     9540000  afg   
4      Afghanistan  1964                0.0     9740000  afg   
...            ...   ...                ...         ...  ...   
11635     Zimbabwe  2015         12800000.0    13800000  zwe   
11636     Zimbabwe  2016         12900000.0    14000000  zwe   
11637     Zimbabwe  2017         14100000.0    14200000  zwe   
11638     Zimbabwe  2018         12900000.0    14400000  zwe   
11639     Zimbabwe  2019         13200000.0    14600000  zwe   

       CELL_PHONE_PER_PERSON  
0                   0.000000  
1                   0.000000  
2                   0.000000  
3                   0.000000  
4                   0.000000  
...                      ...  
11635               0.927536  
11636               0.921429  
11637               0.992958  
11638               0.895833  
11639               0.904110  

[11640 rows x 6 columns]

In [104]:
B = CP_P_GC.loc[CP_P_GC['Country'] == 'United States' ]
B = B.loc[B['year'] == 2017 ]
B

Country  year  cell_phones_total  population  Geo  \
11157  United States  2017        400000000.0   325000000  USA   

       CELL_PHONE_PER_PERSON  
11157               1.230769

In [95]:
CP_P_GC['Geo'] = CP_P_GC['Geo'].str.upper()

In [96]:
CP_P_GC.head()

Country  year  cell_phones_total  population  Geo  \
0  Afghanistan  1960                0.0     9000000  AFG   
1  Afghanistan  1961                0.0     9170000  AFG   
2  Afghanistan  1962                0.0     9350000  AFG   
3  Afghanistan  1963                0.0     9540000  AFG   
4  Afghanistan  1964                0.0     9740000  AFG   

   CELL_PHONE_PER_PERSON  
0                    0.0  
1                    0.0  
2                    0.0  
3                    0.0  
4                    0.0

In [97]:
CP_P_GC.describe()

year  cell_phones_total    population  CELL_PHONE_PER_PERSON
count  11640.000000       1.164000e+04  1.164000e+04           11640.000000
mean    1989.500000       7.875722e+06  2.710307e+07               0.235536
std       17.318846       5.572496e+07  1.086187e+08               0.433303
min     1960.000000       0.000000e+00  4.380000e+03               0.000000
25%     1974.750000       0.000000e+00  1.117500e+06               0.000000
50%     1989.500000       0.000000e+00  5.100000e+06               0.000000
75%     2004.250000       5.615000e+05  1.590000e+07               0.233667
max     2019.000000       1.730000e+09  1.430000e+09               2.126068

In [105]:
CP_P_GC.describe(exclude=[np.number]) 

Country    Geo
count     11640  11640
unique      194    194
top     Tunisia    SMR
freq         60     60

In [122]:
Y_2017 = CP_P_GC.loc[CP_P_GC['year'] == 2017]

In [126]:
l = pd.DataFrame(Y_2017.groupby('Country')['cell_phones_total'].sum()).reset_index()
l.columns=("COUNTRY","CELL_PHONES")
k = l.sort_values(by = 'CELL_PHONES',ascending=False)
k= k.set_index('COUNTRY')
k[:5]

CELL_PHONES
COUNTRY                    
China          1.470000e+09
India          1.170000e+09
Indonesia      4.350000e+08
United States  4.000000e+08
Russia         2.270000e+08